### PR TITLE
[MAINTENANCE] ExpectationSuite.name is source of truth

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/expectation_suite.py [hunk 1]
```diff
@@ -144,8 +144,12 @@ def __init__(  # noqa: PLR0913
...
 
     @property
-    def name(self) -> str:
-        return self.expectation_suite_name
+    def expectation_suite_name(self) -> str:
+        return self.name
+
+    @expectation_suite_name.setter
+    def expectation_suite_name(self, value) -> None:
+        self.name = value
```
Comment (cdkini) [inline great_expectations/core/expectation_suite.py:152]: Can we just remove this since we're open to breaking changes?

**Review by cdkini — APPROVED** [review]

File: tests/core/test_expectation_suite.py [hunk 2]
```diff
@@ -598,6 +598,14 @@ def test_expectation_save_callback_can_come_from_any_copy_of_a_suite(
             == updated_column_name
         )
 
+    @pytest.mark.unit
+    def test_expectation_suite_name_can_be_updated(self, empty_cloud_context_fluent):
```
Comment (cdkini) [inline tests/core/test_expectation_suite.py:602]: Do we need the context here?

**Review by cdkini — CHANGES_REQUESTED** [review]
One minor change

File: tests/core/test_expectation_suite.py (hunk 2 above)
Comment (joshua-stauffer) [reply tests/core/test_expectation_suite.py:602]: yes - a suite can't be instantiated outside the context of a `DataContext`

File: great_expectations/core/expectation_suite.py (hunk 1 above)
Comment (joshua-stauffer) [reply great_expectations/core/expectation_suite.py:152]: yes! was going to scope that to a different PR, since its significantly more involved

**Review by cdkini — APPROVED** [review]

**Review by tyler-hoffman — APPROVED** [review]

**Pull request merged**